examples/large_models/llama-fairscale/Readme.md

```diff
@@ -0,0 +1,86 @@
+# Loading large Huggingface models with constrained resources using accelerate
+
+
+This document briefs on serving the [Llama 2](https://huggingface.co/meta-llama) as presented in the original [Llama repo](https://github.com/facebookresearch/llama/tree/main) using FairScale Tensor Parallel. It basically, is relying on Tensor Parallel layers that build the model, more of Megatron Style. In the following, we show the steps how to serve the 7-70B  model with Torchserve.
+
+### Step 1: Download model
+
+Login into huggingface hub with token by running the below command
+
+```bash
+huggingface-cli login
+```
+paste the token generated from huggingface hub. Make sure `use_auth_token=True` is in [Download script](../utils/Download_model.py).
+
+```bash
+python ../utils/Download_model.py --model_name meta-llama/Llama-2-7b
+```
+The script prints the path where the model is downloaded as below.
+
+`model/models--meta-llama--Llama-2-7b/snapshots/365ffa8f1a6c455d3e2028ae658236b4b85ba824`
+
+The downloaded model is around 14GB.
+
+### Step 2: Configure the settings in the model-config.yaml
+
+The current setting has been tested on A100 GPUs, this may not work on GPUs with memories smaller than in A100 for now.
+
+For sering 7B model set the `nproc-per-node: 1`, for 13B `nproc-per-node: 2` and for 70B `nproc-per-node: 8`.
+
+```bash
+
+#frontend settings
+minWorkers: 1
+maxWorkers: 1
+maxBatchDelay: 200
+responseTimeout: 300
+parallelType: "tp"
+deviceType: "gpu"
+
+torchrun:
+    nproc-per-node: 1
```

Comment:
Mention in the readme to change this value to 2 for 13b and 8 for 70b param model to match the MP size of the Meta Llama 2 weights